packages/platform-sdk-ark/src/services/ledger.ts

```diff
@@ -41,4 +58,42 @@ export class LedgerService implements Contracts.LedgerService {
 	public async signMessage(path: string, payload: Buffer): Promise<string> {
 		return this.#transport.signMessageWithSchnorr(path, payload);
 	}
+
+	public async scan(): Promise<Contracts.WalletData[]> {
+		const pageSize = 5;
+		let page = 0;
+		const slip44 = this.#config.get<number>("network.crypto.slip44");
+
+		let wallets: Contracts.WalletData[] = [];
+		let collection: WalletDataCollection;
+		do {
+			const addresses: string[] = [];
+
+			for (const index of createRange(page, pageSize)) {
+				const path: string = formatLedgerDerivationPath({ coinType: slip44, account: index });
+				const publicKey = await this.getPublicKey(path);
```

Comment:
This is actually interacting with the nano ledger. It might eventually have a failure or 2. Do we need a retry policy here?

Reply:
How often do those errors occur? Don't think any of our products ever had a retry mechanism so far.